core/Azure.Mcp.Core/src/Services/Http/HttpClientFactoryConfigurator.cs

```diff
@@ -0,0 +1,146 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using System.Runtime.Versioning;
+using Azure.Mcp.Core.Areas.Server.Options;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+
+namespace Azure.Mcp.Core.Services.Http;
+
+public static class HttpClientFactoryConfigurator
+{
+    private static readonly string s_version;
+    private static readonly string s_framework;
+    private static readonly string s_platform;
+
+    static HttpClientFactoryConfigurator()
+    {
+        var assembly = typeof(HttpClientService).Assembly;
+        s_version = assembly.GetCustomAttribute<AssemblyFileVersionAttribute>()?.Version ?? "unknown";
+        s_framework = assembly.GetCustomAttribute<TargetFrameworkAttribute>()?.FrameworkName ?? "unknown";
+        s_platform = RuntimeInformation.OSDescription;
+    }
+
+    public static IHttpClientBuilder AddConfiguredHttpClient(this IServiceCollection services)
+    {
+        ArgumentNullException.ThrowIfNull(services);
+
+        // for curious eyes, if one doesn't use named clients, AddHttpClient() doesn't return a builder that can be further configured
+        var builder = services.AddHttpClient("default");
+
+        builder.ConfigureHttpClient((serviceProvider, client) =>
+        {
+            var httpClientOptions = serviceProvider.GetService<IOptions<HttpClientOptions>>()?.Value ?? new HttpClientOptions();
+            client.Timeout = httpClientOptions.DefaultTimeout;
+
+            var transport = serviceProvider.GetService<IOptions<ServiceStartOptions>>()?.Value.Transport ?? TransportTypes.StdIo;
+            client.DefaultRequestHeaders.UserAgent.ParseAdd(BuildUserAgent(transport));
```

Comment:
The delegate parameter sent to `ConfigureHttpClient` will be invoked every time the associated `HttpClient` needs to be constructed. This means we'll run `BuildUserAgent` to recreate the same string many times. While not a huge deal, it's somewhat inefficiently recurring string allocation. This is partly why past implementations created the user agent string was set to static fields.

Reply:
This could also be a follow up PR. I'd recommend chatting with @srnagar.

Reply:
Yes. This is exactly the issue. We need the startup option which is why it's injecting those services.